budgetportal/openspending.py

```diff
@@ -108,18 +135,18 @@ def aggregate(self, cuts=None, drilldowns=None):
             params['cut'] = "|".join(cuts)
         if drilldowns is not None:
             params['drilldown'] = "|".join(drilldowns)
-        aggregate_url = self.cube_url + 'aggregate/'
-        aggregate_result = self.session.get(aggregate_url, params=params)
-        logger.info(
-            "request %s with query %r took %dms",
-            aggregate_result.url,
-            pformat(params),
-            aggregate_result.elapsed.microseconds / 1000
-        )
-        aggregate_result.raise_for_status()
-        if aggregate_result.json()['total_cell_count'] > PAGE_SIZE:
-            raise Exception("More cells than expected - perhaps we should start paging")
-        return aggregate_result.json()
+        url = self.cube_url + 'aggregate/'
+        req = requests.Request('GET', url, params=params)
+        prepped_req = self.session.prepare_request(req)
+        prepped_req.params = params  # <- so that we can access params for logging in aggregate()
+        if csv:
+            csv_url = reverse('csv')
+            csv_url += '?api_url=' + urllib.quote(prepped_req.url)
```

Comment:
rather make this method always return the OpenSpending API URL:
- drop the params dict logging - we can urldecode the URL if we really wanna see the params
- urlencode the OpenSpending URL and construct the relative URL in the view code - this module method shouldn't be aware of that stuff

Reply:
Aggregate() utilises this function to produce it's URL, so if we always return the csv then the other links won't work, or am I missing something?